subsys/bootloader/bl_boot/bl_boot.c

```diff
@@ -105,6 +105,14 @@ void bl_boot(const struct fw_info *fw_info)
 	__set_MSP(vector_table[0]);
 	__set_PSP(0);
 
+#ifdef CONFIG_CPU_CORTEX_M_HAS_SPLIM
```

Comment:
Probably here you should #ifdef on whether the Stack-overflow feature is ON.
Cause the CPU might have the splim registers but might not use them.
This will save you some bytes (which, I guess is crucial on BL0)

Reply:
This is fine.